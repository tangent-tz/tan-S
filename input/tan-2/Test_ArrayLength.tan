main {
    var a1 := [1, 2, 3, 4, 5]; 
    var len := length a1; 
    print a1 \n; 
    print len \n; 
    print \n; 
    
   
    var a2 := [1.2, 4.3+21.2, 3.86409]; 
    len := length a2;
    print a2 \n; 
    print len \n; 
    print \n;     
    
    
    var a3 := ["hello", "world"]; 
    len := length a3;
    print a3 \n; 
    print len \n;
    print \n; 
    
    
    var a4 := ['a', 'l', 'm', 'n', 'o', 'p']; 
    len := length a4;
    print a4 \n; 
    print len \n;
    print \n; 
    
    
    var a5 := [true]; 
    len := length a5; 
    print a5 \n; 
    print len \n;   
    print \n; 
    
    
    print length [true, false, true, false, true] \n; 
    print length [true, false, true, false, true] * 10 \n;
    
    
    print length a5 \n; 
    print length a5 + length a4 \n; 
    
    print length new [int](11) \n; 
    print -length new[int](11) \n; 
    print (length new [int](11)) \n;
     

}