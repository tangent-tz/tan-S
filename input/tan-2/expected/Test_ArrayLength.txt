[ 1, 2, 3, 4, 5 ]
5

[ 1.200000, 25.500000, 3.864090 ]
3

[ hello, world ]
2

[ a, l, m, n, o, p ]
6

[ true ]
1

5
50
1
7
11
-11
11
